sig-cli/charter.md

```diff
@@ -11,7 +11,86 @@ libraries to interface with Kubernetes API's.
 
 ### In scope
 
-SIG CLI [README]
+#### Transparently Exposing APIs through Declarative and Imperative Workflows
+
+The scope of CLI Tools focuses on enabling declarative and imperative workflows
+for invoking kubernetes APIs and authoring Resource Config.  Tools provide
+commands for both generalized (e.g. create resource from Resource Config) tasks and
+specialized (e.g. drain a node, exec into a container) tasks.
+
+It is the philosophy of the tools developed in SIG CLI to facilitate working
+directly with the Kubernetes APIs and Kubernetes style Resources, and to the
+extend possible, provide a transparent experience for how commands map to
+Kubernetes APIs and Resources.
+
+Building tools that obfuscate the underlying Resources and APIs (e.g. through
+generalized templating or DSLs) is an anti-goal of SIG CLI.  Notable examples
+of commands that violate this principle: `run`, `expose`, `autoscale`
+
+#### Specific Types of Functionality
+
+SIG CLI develops tools within the Kubernetes project including but not limited
+to the following:
+
+- Invoking Kubernetes APIs:
+  - Resource APIs - e.g. `create`, `replace`, `delete`, `patch`, `get`, etc
+  - SubResource APIs - e.g. `exec`, `attach`, `logs`, `scale`, etc
+  - Discovery Service - e.g. `api-versions`, `api-resources
+  - Version - e.g. `version`
+  - OpenAPI - e.g. `explain`
+- Pre and Post processing API Resource Config and API invocation responses
+  - `apply` - creates a patch from Resource Config by performing a 3-way diff
+  - `get` - formats API responses into columns
+- Aggregating multiple API invocations and post processing the results
+  - `describe` - performs multiple calls and aggregates results
+  - `run --expose` - creates multiple resources
+- Collapsing multiple manual steps into a command
+  - `edit` - collapses `get`, edit a file, `patch` into a single command
+  - `cp` - collapses manual steps for copy a file out of a container
+- Generating Kubernetes Resource Config locally or creating Resources remotely
+  - From commands + arguments + flags - e.g. `create configmap`, `run`, `expose`
+  - From declarative files - Kustomization `configMapGenerator`, `secretMapGenerator`
+- Transforming Kubernetes Resource Config locally or patching remotely
+  - From commands + arguments + flags - e.g. `annotate`, `set image`, `patch`
```

Comment:
s/set image/set
there are a few flavors

Reply:
yeah, there are a bunch.  This is just a sampling (e.g. also `label`, `scale`, etc)